plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/CodeWhispererCodeScanManager.kt

```diff
@@ -289,19 +290,25 @@ class CodeWhispererCodeScanManager(val project: Project) {
         } finally {
             // After code scan
             afterCodeScan(scope)
-            launch {
-                val duration = (Instant.now().toEpochMilli() - startTime).toDouble()
-                CodeWhispererTelemetryService.getInstance().sendSecurityScanEvent(
-                    CodeScanTelemetryEvent(
-                        codeScanResponseContext,
-                        duration,
-                        codeScanStatus,
-                        codeScanResponseContext.payloadContext.srcPayloadSize.toDouble() ?: 0.0,
-                        connection,
-                        scope
+            if (!(
+                    scope == CodeWhispererConstants.CodeAnalysisScope.FILE &&
+                        (!language.isAutoFileScanSupported() || language == CodeWhispererPlainText.INSTANCE)
```

Comment:
This part is hard to read and understand.
Also, i noticed we have 2 language variables, one in CodeScanManager and the other in CodeScanSessionConfig. https://github.com/aws/aws-toolkit-jetbrains/blob/main/plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/sessionconfig/CodeScanSessionConfig.kt#L76

can we merge these 2?

Reply:
I feel we could handle language param earlier and return early if the lang is not supported by codeScan

Reply:
can we handle language here https://github.com/aws/aws-toolkit-jetbrains/blob/main/plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/CodeWhispererCodeScanManager.kt#L149-L163 ?

Reply:
also, i'm not sure the changes you made here will be actually be run given thisl `return@launch` https://github.com/aws/aws-toolkit-jetbrains/blob/main/plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/CodeWhispererCodeScanManager.kt#L236-L245

Reply:
i think for this PR's purpose you can simply add `var isSkipped = false` before try {} and set it to `true` inside `return@launch` and `if (!isSkipped) {.sendTelemetry }`

Reply:
Will amend to this.

Reply:
Added skip telemetry check for better readability.

Reply:
for early language checks. it gets tricky because project scans don't need language check because we need to send every file to backend on the other hand...we check active file for file scans here: https://github.com/aws/aws-toolkit-jetbrains/blob/main/plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/CodeWhispererCodeScanManager.kt#L227. So maybe we can add checks right at event triggers.